x-pack/plugins/endpoint/public/applications/endpoint/store/managing/middleware.ts

```diff
@@ -32,5 +41,15 @@ export const managementMiddlewareFactory: MiddlewareFactory<ManagementListState>
         payload: response,
       });
     }
+    if (action.type === 'userChangedUrl' && hasSelectedHost(state) !== undefined) {
+      const { selected_host: selectedHost } = uiQueryParams(state);
+      if (selectedHost !== undefined) {
+        const response = await coreStart.http.get(`/api/endpoint/metadata/${selectedHost}`);
```

Comment:
Why is the API endpoint called `/endpoint/metadata`? Isn't it a list of hosts? @kevinlog

Reply:
@peluja1012 we talked about a couple different options here besides `/endpoint/endpoints`.  One was `/endpoint/hosts`.  We settled on `/endpoint/metadata` since the information will eventually be more than just host data and will include more data specific to `endpoints`